src/robusta/integrations/jira/client.py

```diff
@@ -160,6 +179,54 @@ def _get_default_project_id(self):
             return default_issue["id"]
         return None
 
+    def _resolve_priority(self, priority_name: str) -> dict:
+        """Resolve Jira priority in this order:
+        1. User configured priority mapping (by name)
+        2. Default Robusta priority names (SEVERITY_JIRA_ID)
+        3. Fallback to standard Jira IDs (1-4)
+
+        Returns:
+            dict: Priority field in format {"name": str} or {"id": str}
+        """
+        # 1. Try user configured priority mapping
+        if hasattr(self, "params") and self.params.priority_mapping:
+            for severity, mapped_name in self.params.priority_mapping.items():
+                if mapped_name == priority_name:
+                    return {"name": mapped_name}
+
+        # 2. Try default Jira Robusta priority names
+        try:
+            # Make a test call to validate the priority name
+            endpoint = "issue"
+            url = self._get_full_jira_url(endpoint)
+            test_payload = self._create_issue_payload({"priority": {"name": priority_name}})
```

Comment:
I think this might add some load to each issue created, because there's an additional API call to Jira for each issue.
In addition, we've seen that Issue creation on Jira might fail, because some Jira projects has mandatory fields

on the `_validate_priorities` method, you get all the available priorities on Jira
I think I would make this check optional (opt in), and only in the Sink constructor 
So, it will only happen once, and will not change the existing behavior for existing users

What do you think?

Reply:
Hello Arik, thanks for taking the time to review this PR.

> I think this might add some load to each issue created, because there's an additional API call to Jira for each issue.

Agree, simplified that.

> In addition, we've seen that Issue creation on Jira might fail, because some Jira projects has mandatory fields

Yes, we also have some for other type of issue, I specifically chosen another one, that not requiring to fill up these fields.

> on the _validate_priorities method, you get all the available priorities on Jira
> I think I would make this check optional (opt in), and only in the Sink constructor
> So, it will only happen once, and will not change the existing behavior for existing users

My idea would be that it would only be running once during the rubusta start and sink constructor, I think it does this now, and also it will log it when the user has custom mappings defined. I've added another check to print out only in debug mode.

Let me please know if that will work. Thanks.